lib/rex/proto/ldap/server.rb

```diff
@@ -149,76 +156,126 @@ def dispatch_request(cli, data)
         #
         # Default LDAP request dispatcher
         #
-        # @param cli [Rex::Socket::Tcp, Rex::Socket::Udp] Client sending the request
+        # @param client [Rex::Socket::Tcp, Rex::Socket::Udp] Client sending the request
         # @param data [String] raw LDAP request data
-        def default_dispatch_request(cli, data)
-          return if data.strip.empty?
+        def default_dispatch_request(client, data)
+          return if data.strip.empty? || data.strip.nil?
+
+          state = {}
+          post_pdu = false
+          processed_pdu_data = {}
+          state[client] = {
+            name: "#{client.peerhost}:#{client.peerport}",
+            ip: client.peerhost,
+            port: client.peerport,
+            service_name: 'ldap'
+          }
 
           data.extend(Net::BER::Extensions::String)
           begin
             pdu = Net::LDAP::PDU.new(data.read_ber!(Net::LDAP::AsnSyntax))
-            wlog("LDAP request has remaining data: #{data}") if !data.empty?
-            resp = case pdu.app_tag
-                   when Net::LDAP::PDU::BindRequest # bind request
-                     cli.authenticated = true
-                     encode_ldap_response(
-                       pdu.message_id,
-                       Net::LDAP::ResultCodeSuccess,
-                       '',
-                       '',
-                       Net::LDAP::PDU::BindResult
-                     )
-                   when Net::LDAP::PDU::SearchRequest # search request
-                     if cli.authenticated
-                       # Perform query against some loaded LDIF structure
-                       treebase = pdu.search_parameters[:base_object].to_s
-                       # ... search, build packet, send to client
-                       encode_ldap_response(
-                         pdu.message_id,
-                         Net::LDAP::ResultCodeNoSuchObject, '',
-                         Net::LDAP::ResultStrings[Net::LDAP::ResultCodeNoSuchObject],
-                         Net::LDAP::PDU::SearchResult
-                       )
-                     else
-                       service.encode_ldap_response(pdu.message_id, 50, '', 'Not authenticated', Net::LDAP::PDU::SearchResult)
-                     end
-                   when Net::LDAP::PDU::UnbindRequest
-                     nil # close client, no response can be sent over unbound comm
-                   else
-                     service.encode_ldap_response(
-                       pdu.message_id,
-                       Net::LDAP::ResultCodeUnwillingToPerform,
-                       '',
-                       Net::LDAP::ResultStrings[Net::LDAP::ResultCodeUnwillingToPerform],
-                       Net::LDAP::PDU::SearchResult
-                     ) end
-            resp.nil? ? cli.close : send_response(cli, resp)
+            vprint_status("LDAP request data remaining: #{data}") unless data.empty?
```

Comment:
No `*print_*` methods should be accessed in `Rex::Proto` namespace.  These methods are only valid in the `Msf` core mixins.

```suggestion
            wlog("LDAP request data remaining: #{data}") unless data.empty?
```